### PR TITLE
Fix init options

### PR DESCRIPTION
**Pull request by wyuenho**
The last time I updated the init options, the `:initialization-options` lambda appeared to work, but that seems to have stopped working. The reason is the init options were sent under a `jedi` key when jls doesn't accept it, so the init options have no effect. This PR fixes this issue, along with bringing the init options up to date.

### CODE REVIEW

**Comment by wyuenho** [comment]
@fredcamps Mind taking a look at this? I think it's quite important to get them merged ASAP as the init options don't work anymore.

File: lsp-jedi.el [hunk 1]
```diff
@@ -32,8 +32,7 @@
 (defgroup lsp-jedi nil
   "LSP support for Python, using Jedi Python Language Server."
   :group 'lsp-mode
-  :link '(url-link "https://github.com/fredcamps/lsp-jedi")
-  :package-version '(lsp-mode . "6.0"))
```
Comment (fredcamps) [inline lsp-jedi.el:36]: why did you remove it?

File: lsp-jedi.el [hunk 2]
```diff
@@ -106,23 +105,59 @@ autocompletion performance but loses goto definition."
...
+  :type 'lsp-string-vector
+  :group 'lsp-jedi)
 
 (lsp-register-custom-settings
  '(("jedi.enable" lsp-jedi-enable)
    ("jedi.startupMessage" lsp-jedi-startup-message)
    ("jedi.markupKindPreferred" lsp-jedi-markup-kind-preferred)
    ("jedi.trace.server" lsp-jedi-trace-server)
    ("jedi.jediSettings.autoImportModules" lsp-jedi-auto-import-modules)
+   ("jedi.jediSettings.caseInsensitiveCompletion" lsp-jedi-case-insensitive-completion t)
```
Comment (fredcamps) [inline lsp-jedi.el:148]: I think that case insensitive completion must be have default value `nil`

**Review by fredcamps — CHANGES_REQUESTED** [review]
thanks @wyuenho, I made some request changes and commentaries

File: lsp-jedi.el (hunk 2 above)
Comment (wyuenho) [reply lsp-jedi.el:148]: It defaults to True on [`jedi-language-server`](https://github.com/pappasam/jedi-language-server/blob/main/jedi_language_server/initialization_options.py#L48) and [coc-jedi](https://github.com/pappasam/coc-jedi/blob/main/package.json#L89).

File: lsp-jedi.el (hunk 1 above)
Comment (wyuenho) [reply lsp-jedi.el:36]: Because you used the lsp script to generate this group, and that script is only meant to be used for the lsp project. This is also not a group introduced by lsp-mode, it's a group introduced by lsp-jedi. Since it has always been here, there's no need to write which version you've introduced it.

**Comment by wyuenho** [comment]
@fredcamps I've responded to your comments

**Review by fredcamps — APPROVED** [review]
